### PR TITLE
CB-17923 Extend NodeServicesCheckerConclusionStep to check all servic…

### PR DESCRIPTION
**Pull request by lturcsanyi**
…e statuses

Improve and extract conclusion messages to properties file
Enhance salt checker conclusion step error handling

See detailed description in the commit message.

### CODE REVIEW

File: common/src/main/resources/messages/messages.properties [hunk 1]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
 aws.variant.migration.failed=Failed to migrate the cloud formation template
 
 authorization.doc=Please read documentation about user management in official doc: https://docs.cloudera.com/management-console/cloud/user-management/topics/mc-managing-user-access.html. More specifically, you can read about roles here: https://docs.cloudera.com/management-console/cloud/user-management/topics/mc-understanding-roles-resource-roles.html. If you need more detailed information about permissions, you need to check underlying service of user management.
+
+conclusion.node.status.monitor.unreachable=Node status monitor is unreachable. The possible reasons are: the cluster doesnt support the node status monitor service, or the network configuration is not CDP compatible, or some required components are in bad health. Please ssh into the cluster and run cdp doctor.
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:592]: spelling error (doesnt)

File: common/src/main/resources/messages/messages.properties [hunk 2]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
...
+conclusion.node.status.monitor.failed=There are unhealthy services on nodes: {0}. Please check the instances on your cloud provider for further details.
+conclusion.node.status.monitor.failed.details=There are unhealthy services on nodes: {0}
+conclusion.salt.collect.unreachable.found=Unreachable nodes: {0}. We detected that cluster members can?t communicate with each other. Please validate if all cluster members are available and healthy through your cloud provider.
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:595]: spelling issue (can?t)

File: common/src/main/resources/messages/messages.properties [hunk 3]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
...
+conclusion.salt.collect.unreachable.found.details=Unreachable salt minions based on salt ping: {0}
+conclusion.salt.collect.unreachable.failed=Collecting unreachable nodes failed. Please validate if all cluster members are available and healthy through your cloud provider.
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:597]: "through your cloud provider" maybe "on your cloud provider" is better, we use that phrase everywhere else

File: common/src/main/resources/messages/messages.properties [hunk 4]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
...
+conclusion.salt.collect.unreachable.failed.details=Collecting unreachable nodes failed, reason: {0}
+conclusion.salt.master.services.unhealthy=There are unhealthy services on master node: {0}. Please check the instances on your cloud provider for further details.
+conclusion.salt.master.services.unhealthy.details=Unhealthy services on master: {0}
+conclusion.salt.minions.unreachable=Unreachable nodes: {0}. We detected that cluster members can?t communicate with each other. Please validate if all cluster members are available and healthy through your cloud provider.
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:601]: can?t

File: common/src/main/resources/messages/messages.properties [hunk 5]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
...
+conclusion.salt.minions.unreachable.details=Unreachable salt minions based on node status service: {0}
+conclusion.network.ccm.not.accessible=CCM is not accessible from node {0}. Please check network settings!
+conclusion.network.ccm.not.accessible.details=CCM health status is {0} for node {1}
+conclusion.network.cloudera.com.not.accessible=Cloudera.com is not accessible from node: {0}. Please check network settings!
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:605]: I think we shouldn't use excalmation marks in notifications

File: common/src/main/resources/messages/messages.properties [hunk 6]
```diff
@@ -588,3 +588,25 @@ aws.variant.migration.successful=Successfully migrated the cloud formation templ
...
+conclusion.network.cloudera.com.not.accessible.details=Cloudera.com accessibility status is {0} for node {1}
```
Comment (bbihari) [inline common/src/main/resources/messages/messages.properties:606]: cloudera.com

**Review by bbihari — APPROVED** [review]

**Review by doktoric — APPROVED** [review]

**Pull request merged**